drivers/storage/ide/uniata/id_probe.cpp

```diff
@@ -1967,16 +1971,24 @@ UniataClaimLegacyPCIIDE(
     if (!resourceList) {
         KdPrint2((PRINT_PREFIX "!resourceList\n"));
         status = STATUS_INSUFFICIENT_RESOURCES;
+#ifdef __REACTOS__
+        goto del_do;
+#else
```

Comment:
Why "move" this block, instead of simply updating it here?

Reply:
So you think moving the (now placed-at-the-end) new cleanup code back there, could be better?
Would the extra `ExFreePool(oldResList)` call be needed there then?

Reply:
I feel its much cleaner to have the cleaner code at the bottom of the function. Jumping randomly to the middle of the function to cleanup is not "pretty" tbh. 
The code was originally changed without the `#ifdef`s so it looked much cleaner. Now its just ugly, but I guess we can't avoid it. Hopefully the UniATA maintainer approves these changes so we can remove the `#ifdef`

> Would the extra ExFreePool(oldResList) call be needed there then?

Yes if the HalAssignSlotResources  fails we must still cleanup the "old" allocated resource list. The new resource list will have be cleaned up by HalAssignSlotResources but the old resource list must be cleaned by us, since its allocated in this function.

Reply:
I do understand your motivation not wanting to not jump into the middle of the func for cleanup. A good rule when coding is to "not surprise other developers", e.g.: with your code-flow.
Resolved.

Reply:
> not "pretty"

I agree for project own code, but, for upstream code, minimal diff is preferable.

> UniATA maintainer approves

That would be great, yet I do not expect it: no answer from him for a long time...

Reply:
Alright, Alright. I moved it back